sdk/core/azure-core/src/main/java/com/azure/android/core/util/events/EventHandler.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.util.events;
+
+/**
+ * A simple event handler.
+ *
+ * @param <E> The event to handle.
+ */
+@FunctionalInterface
```

Comment:
Drop the functional interface annotation (as discussed it requires API level 24), and validate why the build system didn't pick this up.